portfolio/src/main/webapp/script.js

```diff
@@ -12,27 +12,36 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
+let key_to_be_deleted = 0;
```

Comment:
Using a mutable global variable `key_to_be_deleted` leads to a bug where the wrong comment is deleted. (Every button deletes the last comment in the list.) Also in general, mutable globals make code harder to read.

I'm not sure what the best thing to do instead is, because I don't know javascript coding conventions.